File: decompiled/sources/com/hjq/bar/style/CommonBarStyle.java
package com.hjq.bar.style;

import android.app.Activity;
import android.content.Context;
import android.content.pm.PackageManager;
import android.text.TextUtils;
import android.util.TypedValue;
import android.view.View;
import android.widget.TextView;
import androidx.core.view.GravityCompat;
import com.hjq.bar.ITitleBarStyle;

public abstract class CommonBarStyle implements ITitleBarStyle {
    public int getLeftIconGravity(Context context) {
        return 8388611;
    }

    public int getLeftIconHeight(Context context) {
        return 0;
    }

    public int getLeftIconWidth(Context context) {
        return 0;
    }

    public CharSequence getLeftTitle(Context context) {
        return "";
    }

    public int getLeftTitleStyle(Context context) {
        return 0;
    }

    public int getLineSize(Context context) {
        return 1;
    }

    public int getRightIconGravity(Context context) {
        return GravityCompat.END;
    }

    public int getRightIconHeight(Context context) {
        return 0;
    }

    public int getRightIconWidth(Context context) {
        return 0;
    }

    public CharSequence getRightTitle(Context context) {
        return "";
    }

    public int getRightTitleStyle(Context context) {
        return 0;
    }

    public int getTitleIconGravity(Context context) {
        return GravityCompat.END;
    }

    public int getTitleIconHeight(Context context) {
        return 0;
    }

    public int getTitleIconWidth(Context context) {
        return 0;
    }

    public int getTitleStyle(Context context) {
        return 0;
    }

    public boolean isLineVisible(Context context) {
        return true;
    }

    public TextView createTitleView(Context context) {
        TextView newTitleView = newTitleView(context);
        newTitleView.setGravity(16);
        newTitleView.setFocusable(true);
        newTitleView.setSingleLine();
        newTitleView.setEllipsize(TextUtils.TruncateAt.MARQUEE);
        newTitleView.setMarqueeRepeatLimit(-1);
        newTitleView.setSelected(true);
        return newTitleView;
    }

    public TextView newTitleView(Context context) {
        return new TextView(context);
    }

    public TextView createLeftView(Context context) {
        TextView newLeftView = newLeftView(context);
        newLeftView.setGravity(16);
        newLeftView.setFocusable(true);
        newLeftView.setSingleLine();
        newLeftView.setEllipsize(TextUtils.TruncateAt.END);
        return newLeftView;
    }

    public TextView newLeftView(Context context) {
        return new TextView(context);
    }

    public TextView createRightView(Context context) {
        TextView newRightView = newRightView(context);
        newRightView.setGravity(16);
        newRightView.setFocusable(true);
        newRightView.setSingleLine();
        newRightView.setEllipsize(TextUtils.TruncateAt.END);
        return newRightView;
    }

    public TextView newRightView(Context context) {
        return new TextView(context);
    }

    public View createLineView(Context context) {
        return new View(context);
    }

    public int getChildHorizontalPadding(Context context) {
        return (int) TypedValue.applyDimension(1, 12.0f, context.getResources().getDisplayMetrics());
    }

    public int getChildVerticalPadding(Context context) {
        return (int) TypedValue.applyDimension(1, 15.0f, context.getResources().getDisplayMetrics());
    }

    public CharSequence getTitle(Context context) {
        if (!(context instanceof Activity)) {
            return "";
        }
        CharSequence title = ((Activity) context).getTitle();
        if (TextUtils.isEmpty(title)) {
            return "";
        }
        try {
            PackageManager packageManager = context.getPackageManager();
            return !title.toString().equals(packageManager.getPackageInfo(context.getPackageName(), 0).applicationInfo.loadLabel(packageManager).toString()) ? title : "";
        } catch (PackageManager.NameNotFoundException unused) {
            return "";
        }
    }

    public float getTitleSize(Context context) {
        return TypedValue.applyDimension(2, 16.0f, context.getResources().getDisplayMetrics());
    }

    public float getLeftTitleSize(Context context) {
        return TypedValue.applyDimension(2, 14.0f, context.getResources().getDisplayMetrics());
    }

    public float getRightTitleSize(Context context) {
        return TypedValue.applyDimension(2, 14.0f, context.getResources().getDisplayMetrics());
    }

    public int getTitleIconPadding(Context context) {
        return (int) TypedValue.applyDimension(1, 2.0f, context.getResources().getDisplayMetrics());
    }

    public int getLeftIconPadding(Context context) {
        return (int) TypedValue.applyDimension(1, 2.0f, context.getResources().getDisplayMetrics());
    }

    public int getRightIconPadding(Context context) {
        return (int) TypedValue.applyDimension(1, 2.0f, context.getResources().getDisplayMetrics());
    }
}

File: decompiled/sources/com/hjq/bar/style/CommonBarStyle.kt
package com.hjq.bar.style

import android.app.Activity
import android.content.Context
import android.content.pm.PackageManager
import android.text.TextUtils
import android.util.TypedValue
import android.view.View
import android.widget.TextView
import androidx.core.view.GravityCompat
import com.hjq.bar.ITitleBarStyle

abstract class CommonBarStyle : ITitleBarStyle {
    override fun getLeftIconGravity(context: Context): Int = 8388611

    override fun getLeftIconHeight(context: Context): Int = 0

    override fun getLeftIconWidth(context: Context): Int = 0

    override fun getLeftTitle(context: Context): CharSequence = ""

    override fun getLeftTitleStyle(context: Context): Int = 0

    override fun getLineSize(context: Context): Int = 1

    override fun getRightIconGravity(context: Context): Int = GravityCompat.END

    override fun getRightIconHeight(context: Context): Int = 0

    override fun getRightIconWidth(context: Context): Int = 0

    override fun getRightTitle(context: Context): CharSequence = ""

    override fun getRightTitleStyle(context: Context): Int = 0

    override fun getTitleIconGravity(context: Context): Int = GravityCompat.END

    override fun getTitleIconHeight(context: Context): Int = 0

    override fun getTitleIconWidth(context: Context): Int = 0

    override fun getTitleStyle(context: Context): Int = 0

    override fun isLineVisible(context: Context): Boolean = true

    override fun createTitleView(context: Context): TextView {
        val newTitleView = newTitleView(context)
        newTitleView.gravity = 16
        newTitleView.isFocusable = true
        newTitleView.setSingleLine()
        newTitleView.ellipsize = TextUtils.TruncateAt.MARQUEE
        newTitleView.marqueeRepeatLimit = -1
        newTitleView.isSelected = true
        return newTitleView
    }

    open fun newTitleView(context: Context): TextView = TextView(context)

    override fun createLeftView(context: Context): TextView {
        val newLeftView = newLeftView(context)
        newLeftView.gravity = 16
        newLeftView.isFocusable = true
        newLeftView.setSingleLine()
        newLeftView.ellipsize = TextUtils.TruncateAt.END
        return newLeftView
    }

    open fun newLeftView(context: Context): TextView = TextView(context)

    override fun createRightView(context: Context): TextView {
        val newRightView = newRightView(context)
        newRightView.gravity = 16
        newRightView.isFocusable = true
        newRightView.setSingleLine()
        newRightView.ellipsize = TextUtils.TruncateAt.END
        return newRightView
    }

    open fun newRightView(context: Context): TextView = TextView(context)

    override fun createLineView(context: Context): View = View(context)

    override fun getChildHorizontalPadding(context: Context): Int =
        TypedValue.applyDimension(1, 12.0f, context.resources.displayMetrics).toInt()

    override fun getChildVerticalPadding(context: Context): Int =
        TypedValue.applyDimension(1, 15.0f, context.resources.displayMetrics).toInt()

    override fun getTitle(context: Context): CharSequence {
        if (context !is Activity) {
            return ""
        }
        val title = context.title
        if (TextUtils.isEmpty(title)) {
            return ""
        }
        return try {
            val packageManager = context.packageManager
            if (!title.toString().equals(
                    packageManager.getPackageInfo(context.packageName, 0).applicationInfo
                        .loadLabel(packageManager).toString()
                )
            ) {
                title
            } else {
                ""
            }
        } catch (_: PackageManager.NameNotFoundException) {
            ""
        }
    }

    override fun getTitleSize(context: Context): Float =
        TypedValue.applyDimension(2, 16.0f, context.resources.displayMetrics)

    override fun getLeftTitleSize(context: Context): Float =
        TypedValue.applyDimension(2, 14.0f, context.resources.displayMetrics)

    override fun getRightTitleSize(context: Context): Float =
        TypedValue.applyDimension(2, 14.0f, context.resources.displayMetrics)

    override fun getTitleIconPadding(context: Context): Int =
        TypedValue.applyDimension(1, 2.0f, context.resources.displayMetrics).toInt()

    override fun getLeftIconPadding(context: Context): Int =
        TypedValue.applyDimension(1, 2.0f, context.resources.displayMetrics).toInt()

    override fun getRightIconPadding(context: Context): Int =
        TypedValue.applyDimension(1, 2.0f, context.resources.displayMetrics).toInt()
}
